### PR TITLE
Add Neutralizing Gas

### PR DESCRIPTION
**Pull request by ghoulslash**
**Implemented:**
- Neutralizing Gas is announced before entry hazards
- Neutralizing Gas suppresses abilities (in this example, Hyper Cutter)
neutralizing_gas2

- Switch-in abilities occur as the Pokemon with Neutralizing Gas switches out
neutralizing_gas

- Neutralizing Gas being suppressed by Simple Beam or gastro Acid:
neutralizing_gas_simple

**NOTE:** the `jumpifbytenotequal gBattlerTarget, sByteFour, BattleScript_NeutralizingGasExitsLoop` in `BattleScript_NeutralizingGasExits` seems unecessary, but comparing directly to `gBattlersCount` caused the battle script to mess up.

### CODE REVIEW

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -5416,7 +5416,7 @@ static void Cmd_sethealblock(void)
 
 static void Cmd_returnatktoball(void)
 {
-    gActiveBattler = gBattlerAttacker;
+    gActiveBattler = gBattlerAttacker;
```
Comment (LOuroboros) [inline src/battle_script_commands.c:5419]:
```suggestion
    gActiveBattler = gBattlerAttacker;
```

File: src/battle_script_commands.c [hunk 2]
```diff
@@ -6240,7 +6249,7 @@ static void Cmd_endlinkbattle(void)
 }
 
 static void Cmd_returntoball(void)
-{
+{
```
Comment (LOuroboros) [inline src/battle_script_commands.c:6252]:
```suggestion
{
```

File: src/battle_script_commands.c [hunk 3]
```diff
@@ -12281,25 +12290,35 @@ static void Cmd_trygetintimidatetarget(void)
 static void Cmd_switchoutabilities(void)
 {
     gActiveBattler = GetBattlerForBattleScript(gBattlescriptCurrInstr[1]);
-
-    switch (GetBattlerAbility(gActiveBattler))
+
```
Comment (LOuroboros) [inline src/battle_script_commands.c:12293]:
```suggestion

```

**Review by LOuroboros — CHANGES_REQUESTED** [review]

**Comment by LOuroboros** [comment]
I can't find information on whether Run Away is supposed to be blocked by Neutralizing Gas or not.
Right now, it's not being affected in this PR :eyes:
%pn_1694

File: src/battle_util.c [hunk 4]
```diff
@@ -5496,11 +5513,51 @@ u8 AbilityBattleEffects(u8 caseID, u8 battler, u16 ability, u8 special, u16 move
...
+    u32 i;
+
+    for (i = 0; i < gBattlersCount; i++)
+    {
+        if (IsBattlerAlive(i) && gBattleMons[i].ability == ABILITY_NEUTRALIZING_GAS && !(gStatuses3[i] & STATUS3_GASTRO_ACID))
+            return TRUE;
+    }
+
+    return FALSE;
+}
```
Comment (LOuroboros) [inline src/battle_util.c:5550]: Instead of adding a function for this, wouldn't it be better to just use `IsAbilityOnField`? :eyes:

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**